webservice/models/webservice_backend.py

```diff
@@ -14,8 +16,19 @@ class WebserviceBackend(models.Model):
     tech_name = fields.Char(required=True)
     protocol = fields.Selection([("http", "HTTP Request")], required=True)
     url = fields.Char(required=True)
-    username = fields.Char()
-    password = fields.Char()
+    auth_type = fields.Selection(
+        selection=[
+            ("none", "Public"),
+            ("user_pwd", "Username & password"),
+            ("api_key", "API Key"),
+        ],
+        default="user_pwd",
+        required=True,
+    )
+    username = fields.Char(auth_type="user_pwd")
+    password = fields.Char(auth_type="user_pwd")
+    api_key = fields.Char(string="API Key", auth_type="api_key")
```

Comment:
I don't understand how this `auth_type` parameter on the filed declaration works and where it comes from ?

Reply:
I'm adding it. See `_valid_field_parameter` override